### ISSUE TITLE
RemoveFile can result in invalid zip file

### ISSUE THREAD

**Issue by billytaylor** [srikanth-lingala/zip4j#107]
Hi,
first of all, thanks for this implementation anyway! The current code assumes that only central directory headers **after** the removed file need their offsets corrected,. I have come across ZIP files (specifically .APK files) which have files early in the central directory (**before** the removed file) but whose offset was (and is)  **after** the removed file, e.g. META-INF/CERT.RSA and META-INF/CERT.SF after a .APK file has been re-signed. This patch handles this case. I hope it helps.

```
--- a/src/main/java/net/lingala/zip4j/tasks/RemoveEntryFromZipFileTask.java
+++ b/src/main/java/net/lingala/zip4j/tasks/RemoveEntryFromZipFileTask.java
@@ -146,15 +146,17 @@ public class RemoveEntryFromZipFileTask extends AsyncZipTask<RemoveEntryFromZipF

   private void updateFileHeadersWithLocalHeaderOffsets(List<FileHeader> fileHeaders, long offsetEndOfCompressedFile,
                                                        long offsetLocalFileHeader, int indexOfFileHeader) {
-    for (int i = indexOfFileHeader; i < fileHeaders.size(); i ++) {
+    for (int i = 0; i < fileHeaders.size(); i ++) {
       FileHeader fileHeader = fileHeaders.get(i);
       long offsetLocalHdr = fileHeader.getOffsetLocalHeader();
       if (fileHeader.getZip64ExtendedInfo() != null && fileHeader.getZip64ExtendedInfo().getOffsetLocalHeader() != -1) {
         offsetLocalHdr = fileHeader.getZip64ExtendedInfo().getOffsetLocalHeader();
       }
-      fileHeader.setOffsetLocalHeader(offsetLocalHdr - (offsetEndOfCompressedFile - offsetLocalFileHeader) - 1);
+      if (offsetLocalHdr > offsetLocalFileHeader) {
+        fileHeader.setOffsetLocalHeader(offsetLocalHdr - (offsetEndOfCompressedFile - offsetLocalFileHeader) - 1);
     }
   }
+  }

   private void cleanupFile(boolean successFlag, File zipFile, File temporaryZipFile) throws ZipException {
     if (successFlag) {
```

**Comment by srikanth-lingala** [comment]
Thanks for pointing it out and the fix. I will include it in the next release

**Comment by srikanth-lingala** [comment]
Issue fixed in v2.2.5
